iotdb/src/main/java/org/apache/iotdb/db/query/control/FileReaderManager.java

```diff
@@ -79,27 +83,33 @@ private void clearUnUsedFilesInFixTime() {
 
     executorService.scheduleAtFixedRate(() -> {
       synchronized (this) {
-        for (Map.Entry<String, TsFileSequenceReader> entry : fileReaderMap.entrySet()) {
-          TsFileSequenceReader reader = entry.getValue();
-          int referenceNum = referenceMap.get(entry.getKey()).get();
-
-          if (referenceNum == 0) {
-            try {
-              reader.close();
-            } catch (IOException e) {
-              LOGGER.error("Can not close TsFileSequenceReader {} !", reader.getFileName(), e);
-            }
-            fileReaderMap.remove(entry.getKey());
-            referenceMap.remove(entry.getKey());
-          }
-        }
+        clearMap(unclosedFileReaderMap, unclosedReferenceMap);
+        clearMap(closedFileReaderMap, closedReferenceMap);
       }
     }, 0, examinePeriod, TimeUnit.MILLISECONDS);
   }
 
+  private void clearMap(Map<String, TsFileSequenceReader> readerMap,
+      Map<String, AtomicInteger> refMap) {
+    for (Map.Entry<String, TsFileSequenceReader> entry : readerMap.entrySet()) {
+      TsFileSequenceReader reader = entry.getValue();
+      int referenceNum = refMap.get(entry.getKey()).get();
+
+      if (referenceNum == 0) {
+        try {
+          reader.close();
+        } catch (IOException e) {
+          LOGGER.error("Can not close TsFileSequenceReader {} !", reader.getFileName(), e);
+        }
+        readerMap.remove(entry.getKey());
+        refMap.remove(entry.getKey());
+      }
+    }
+  }
+
   /**
    * Get the reader of the file(tsfile or unseq tsfile) indicated by filePath. If the reader already
-   * exists, just get it from fileReaderMap. Otherwise a new reader will be created.
+   * exists, just get it from closedFileReaderMap. Otherwise a new reader will be created.
```

Comment:
`just get it from closedFileReaderMap`, it may also come from `unclosedFileReaderMap`

Reply:
fixed